play-validations/memory-footprint/src/main/java/com/google/wear/watchface/dfx/memory/DrawableResourceDetails.java

```diff
@@ -225,6 +224,97 @@ private static String getResourceName(InputPackage.PackageFile packageFile) {
         return resourceNameWithExtension;
     }
 
+    /**
+     * Evaluates the memory footprint of a drawable asset. The memory footprint is defined as
+     * number_of_frames * width * height * 4 bytes, or how much memory does storing that resource
+     * take in its uncompressed format.
+     *
+     * @param resource the resource from a watch face package.
+     * @return the memory footprint of that asset file or {@code Optional.empty()} if the file is
+     *     not a drawable asset.
+     * @throws java.lang.IllegalArgumentException when the image cannot be processed.
+     */
+    static Optional<DrawableResourceDetails> fromPackageResource(AndroidResource resource) {
```

Comment:
I don't think `fromPackageFile` is used anymore, except for one test. Is it possible to refactor that test to use `fromPackageResource` and delete the other method? Otherwise, this method should be broken up in some way to avoid duplication, but ideally the old one should go away.

Similarly, `InputPackage.PackageFile` should go away.

Reply:
Thanks, great call. done.